pkg/rpc/server/server.go

```diff
@@ -248,7 +248,7 @@ func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Requ
 			s.log.Info("websocket connection upgrade failed", zap.Error(err))
 			return
 		}
-		resChan := make(chan response.Abstract)
+		resChan := make(chan interface{}) // response.Abstract or []response.Abstract
```

Comment:
How about `json.Marshaler` here? `interface{}` is kinda the last resort, for cases when nothing else works, I think here we can use an interface.

Reply:
I made a `response.AbstructResult` interface for that. The interface has one method `RunForErrors(func (*response.Error))`. Adding `MarshalJSON` to the interface requires explicit method implementation, but marshalling is so trivial for response.Abstract and response.AbstractBatch that it can be done automatically.

Reply:
If it works without it (and it does), we don't. It looks fine with `AbstractResult`, much better than `interface{}`.